lib/loader.js

```diff
@@ -86,7 +86,14 @@ function sassLoader(content) {
       // Since we specified options.sourceMap = path.join(process.cwd(), "/sass.map"); in normalizeOptions,
       // we know that this path is relative to process.cwd(). This is how node-sass works.
       // eslint-disable-next-line no-param-reassign
-      result.map.sources[0] = path.relative(process.cwd(), resourcePath);
+      let stdinIndex = result.map.sources.findIndex(
+        (source) => source.indexOf('stdin') !== -1
+      );
+      stdinIndex = stdinIndex !== -1 ? stdinIndex : 0;
+      result.map.sources[stdinIndex] = path.relative(
+        process.cwd(),
+        resourcePath
+      );
```

Comment:
Ok, thanks for ivnestigation, maybe rewrite this like:
```
if (stdinIndex !== -1) {
  result.map.sources[stdinIndex] = path.relative(
        process.cwd(),
        resourcePath
      );
}
```

This allow to us protect code if stdin was excluded from sources in future

Reply:
I'm not sure I understand the point - the way I've written it, `stdinIndex` will never be -1. If it is, it will be 0.

Reply:
@FractalBoy I am afraid what in future `node-sass`/`sass` can remove `stdin` from `result.map.sources` so better check `stdin` exists in `sources` and then modify

Reply:
Now- If `stdin` not exists in `sources` we break source map, because on `0` index will be path to source

Reply:
I see. I'll make the change.

Reply:
Maybe it would actually be better to remove stdin if it exists and then always add the path?
```
      const stdinIndex = result.map.sources.findIndex(
        (source) => source.indexOf('stdin') !== -1
      );

      const relativeResourcePath = path.relative(
        process.cwd(),
        resourcePath
      );

      if (stdinIndex !== -1) {
        result.map.sources[stdinIndex] = relativeResourcePath;
      } else {
        result.map.sources.push(relativeResourcePath);
      }
```

Reply:
Actually, nevermind, that would probably screw up the source maps since it wouldn't line up correctly.